src/plugins/vis_type_table/public/paginated_table/rows.js

```diff
@@ -65,7 +66,9 @@ export function KbnRows($compile) {
 
         if (column.filterable && contentsIsDefined) {
           $cell = createFilterableCell(contents);
-          $cellContent = $cell.find('[data-cell-content]');
+          // in jest tests 'angular' is using jqLite. In jqLite the method find lookups only by tags.
+          // Because of this, we should change a way how we get cell content so that tests will pass.
+          $cellContent = angular.element($cell[0].querySelector('[data-cell-content]'));
```

Comment:
as an alternative to your solution, we can use a not-so-popular  tag (e.g. `<section>`) for an element with` [data-cell-content]` tag
But I'm ok with this solution too